docs/deployment/getting-started-locally/manage-s3-emulator.md

```diff
@@ -0,0 +1,69 @@
+# Manage S3 Emulator
+
+This document is a step-by-step guide on how to configure, deploy and cleanup [LocalStack](https://localstack.cloud/), within a [kind](https://kind.sigs.k8s.io/) cluster. LocalStack emulates AWS services locally, which allows the `Etcd` cluster to interact with AWS S3. This setup is ideal for local development and testing.
+
+## 00-Prerequisites
+
+Ensure that you have setup the development environment as per the [documentation](../prepare-dev-environment.md).
+
+> **Note:** It is assumed that you have already created kind cluster and the `KUBECONFIG` is pointing to this Kubernetes cluster.
+
+### Installing AWS CLI
+
+To interact with `LocalStack` you must also install the AWS CLI `(version >=1.29.0 or version >=2.13.0)`
+On macOS run:
+
+```bash
+> brew install awscli
+```
+
+For other OS, please check the [AWS CLI installation documentation](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).
+
+## 01-Deploy LocalStack
+
+```bash
+> make deploy-localstack
+```
+
+The above make target will deploy `LocalStack` in the target Kubernetes cluster.
+
+## 02-Setup S3 Bucket
+
+Configure `AWS CLI` to interact with LocalStack by setting the necessary environment variables. This configuration redirects S3 commands to the LocalStack endpoint and provides the required credentials for authentication.
+
+```bash
+> export AWS_ENDPOINT_URL_S3="http://localhost:4566"
+> export AWS_ACCESS_KEY_ID=ACCESSKEYAWSUSER
+> export AWS_SECRET_ACCESS_KEY=sEcreTKey
+> export AWS_DEFAULT_REGION=us-east-2
+```
+
+Create a S3 bucket using the following command:
+```bash
+> aws s3api create-bucket --bucket <bucket-name> --region <region> --create-bucket-configuration LocationConstraint=<region> --acl private
+```
+
+To verify if the bucket has been created, you can use the following command:
+```bash
+> aws s3 ls <bucket-name>/<etcd-name>/v2/
+```
```

Comment:
Until the etcd cluster is created and backup snapshotter is started, the path `<bucket-name>/<etcd-name>/v2/` will not exist. Instead, why not simply replace the command with just `aws s3 ls` to list the buckets and check whether the bucket name exists there?

Reply:
actually the correct thing to use here is `aws s3api head-bucket --bucket <bucket-name>`